flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -383,8 +381,10 @@ protected IcebergSource<RowData> source() {
         .project(TestFixtures.TS_SCHEMA)
         .splitSize(100L)
         .streaming(true)
-        .monitorInterval(Duration.ofMillis(2))
+        .monitorInterval(Duration.ofMillis(100))
         .streamingStartingStrategy(StreamingStartingStrategy.TABLE_SCAN_THEN_INCREMENTAL)
+        // Read in single row batches, to have more batches for testing
+        .flinkConfig(new Configuration().set(SOURCE_READER_FETCH_BATCH_RECORD_COUNT, 1))
```

Comment:
is this needed for fixing test flakiness? if so, we should separate them out and merge ASAP.

Reply:
Nope. This is needed for higher probability for hitting `pauseOrResumeSplits` and friends. Since the fetching is done in batches, having 1 record per batch allows us to have more fetch cycles - bigger probability to run into the blocking/unblocking situations.